src/main.c

```diff
@@ -122,12 +139,16 @@ main(int argc, char **argv)
 
     main_loop = g_main_loop_new(NULL, FALSE);
 
-    client = snapd_client_new();
-    snapd = ds_snapd_helper_new(client);
+    state = g_new0(DsState, 1);
+    state->client = snapd_client_new();
+
+    if (g_getenv ("SNAP") != NULL) {
+        snapd_client_set_socket_path(state->client, "/run/snapd-snap.socket");
```

Comment:
This doesn't seem to work when not running as a snap, so I only set it when we are in a snap - is that expected? Also not sure if checking for `$SNAP` is the best method of determining when running inside a snap.

Reply:
Checking that SNAP or SNAP_NAME are set should be fine here.